Userland/Libraries/LibCore/Account.cpp

```diff
@@ -295,22 +295,20 @@ ErrorOr<void> Account::sync()
     auto new_shadow_file_content = TRY(generate_shadow_file());
 #endif
 
-    // FIXME: mkstemp taking Span<char> makes this code entirely un-AKable.
-    //        Make this code less char-pointery.
-    char new_passwd_name[] = "/etc/passwd.XXXXXX";
-    size_t new_passwd_name_length = strlen(new_passwd_name);
+    auto new_passwd_file = TRY(ByteBuffer::copy("/etc/passwd.XXXXXX"sv.bytes()));
+    auto new_passwd_file_span = Span<char> { reinterpret_cast<char*>(new_passwd_file.data()), new_passwd_file.size() };
 #ifndef AK_OS_BSD_GENERIC
-    char new_shadow_name[] = "/etc/shadow.XXXXXX";
-    size_t new_shadow_name_length = strlen(new_shadow_name);
+    auto new_shadow_file = TRY(ByteBuffer::copy(("/etc/shadow.XXXXXX"sv.bytes())));
+    auto new_shadow_file_span = Span<char> { reinterpret_cast<char*>(new_shadow_file.data()), new_shadow_file.size() };
```

Comment:
I think heap-allocating a byte buffer is not an improvement here. The FIXME seems a little misguided - we can pass the `char[]` values directly into `Core::System::mkstemp` without any issue, and then just `strlen` on the invocations to `Core::System::rename`. That is:

```diff
index 68de38c562..03848dfed1 100644
--- a/Userland/Libraries/LibCore/Account.cpp
+++ b/Userland/Libraries/LibCore/Account.cpp
@@ -295,22 +295,18 @@ ErrorOr<void> Account::sync()
     auto new_shadow_file_content = TRY(generate_shadow_file());
 #endif
 
-    // FIXME: mkstemp taking Span<char> makes this code entirely un-AKable.
-    //        Make this code less char-pointery.
     char new_passwd_name[] = "/etc/passwd.XXXXXX";
-    size_t new_passwd_name_length = strlen(new_passwd_name);
 #ifndef AK_OS_BSD_GENERIC
     char new_shadow_name[] = "/etc/shadow.XXXXXX";
-    size_t new_shadow_name_length = strlen(new_shadow_name);
 #endif
 
     {
-        auto new_passwd_fd = TRY(Core::System::mkstemp({ new_passwd_name, new_passwd_name_length }));
+        auto new_passwd_fd = TRY(Core::System::mkstemp(new_passwd_name));
         ScopeGuard new_passwd_fd_guard = [new_passwd_fd] { close(new_passwd_fd); };
         TRY(Core::System::fchmod(new_passwd_fd, 0644));
 
 #ifndef AK_OS_BSD_GENERIC
-        auto new_shadow_fd = TRY(Core::System::mkstemp({ new_shadow_name, new_shadow_name_length }));
+        auto new_shadow_fd = TRY(Core::System::mkstemp(new_shadow_name));
         ScopeGuard new_shadow_fd_guard = [new_shadow_fd] { close(new_shadow_fd); };
         TRY(Core::System::fchmod(new_shadow_fd, 0600));
 #endif
@@ -324,9 +320,9 @@ ErrorOr<void> Account::sync()
 #endif
     }
 
-    TRY(Core::System::rename({ new_passwd_name, new_passwd_name_length }, "/etc/passwd"sv));
+    TRY(Core::System::rename({ new_passwd_name, strlen(new_passwd_name) }, "/etc/passwd"sv));
 #ifndef AK_OS_BSD_GENERIC
-    TRY(Core::System::rename({ new_shadow_name, new_shadow_name_length }, "/etc/shadow"sv));
+    TRY(Core::System::rename({ new_shadow_name, strlen(new_shadow_name) }, "/etc/shadow"sv));
 #endif
 
     return {};
```